### PR TITLE
Add deprecation for equalto and occursin

### PR DESCRIPTION
**Pull request by ararslan**
These were removed in #26436 without deprecation, so this PR adds a deprecation for them ~~plus a NEWS item~~. It also addresses https://github.com/JuliaLang/julia/pull/26436#discussion_r174888410.

### CODE REVIEW

**Comment by mbauman** [comment]
I'd remove the NEWS — the context there is breaking changes from 0.6.  I think that'd just be confusing for the vast majority of the audience.

File: base/deprecated.jl [hunk 1]
```diff
@@ -1510,6 +1510,12 @@ end
 # Issue #26248
 @deprecate conj(x) x
 
+# PR #26436
+@deprecate equalto(x) isequal(x)
+@deprecate(occursin(x), in(x))
```
Comment (JeffBezanson) [inline base/deprecated.jl]: Why the parens?

File: base/deprecated.jl [hunk 2]
```diff
@@ -1510,6 +1510,12 @@ end
...
+@deprecate_binding EqualTo Base.Fix2 false
+@deprecate_binding OccursIn Base.Fix2 false
```
Comment (JeffBezanson) [inline base/deprecated.jl]: I left in definitions for these actually:

```
const EqualTo = Fix2{typeof(isequal)}
const OccursIn = Fix2{typeof(in)}
```

I'm not sure what their ultimate fate will be but they're pretty harmless for now.

File: base/deprecated.jl (hunk 1 above)
Comment (ararslan) [reply base/deprecated.jl]: Otherwise it parses as `occursin(x) in x`

File: base/deprecated.jl (hunk 2 above)
Comment (ararslan) [reply base/deprecated.jl]: The use of `OccursIn` there will be kind of weird once the `contains` -> `occursin` replacement goes through though.

File: base/deprecated.jl (hunk 1 above)
Comment (JeffBezanson) [reply base/deprecated.jl]: d'oh!

File: base/deprecated.jl (hunk 2 above)
Comment (JeffBezanson) [reply base/deprecated.jl]: Fair enough, but we have to do something. They can't both be deprecated just to `Fix2`, since that won't work.

Comment (ararslan) [reply base/deprecated.jl]: Okay, I can just remove these deprecations and we can probably just silently remove `EqualTo` and `OccursIn` between 0.7 and 1.0.

Comment (mbauman) [reply base/deprecated.jl]: Why not just do `@deprecate_binding EqualTo Base.Fix2{typeof(isequal)} false`?  That seems to work.

Comment (ararslan) [reply base/deprecated.jl]: Do we want to make the internal definitions use `<:Union{typeof(isequal), typeof(==)}`?

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**